### PR TITLE
Spanish JS UI tests for document upload

### PR DESCRIPTION
**Pull request by bartzet**
# Problem
Currently, we have a number of Ruby UI tests for Document upload that test the Spanish language. We would like to move them to JS so that it will reduce the number of Ruby UI tests.

# Solution

1. JS UI tests loop thorough the document upload tests and test Spanish translations too.
2. Remove Ruby UI tests for document upload - scenarios and step definitions.

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [ ] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [ ] Have new automated tests been implemented?
- [ ] Have new manual tests been written down?
- [ ] Have tests passed locally?
- [ ] Have any new strings been translated?

### CODE REVIEW

File: test/js/specs/happy.js [hunk 1]
```diff
@@ -70,151 +71,153 @@ describe('Happy Paths', options, ({driver, pageObjects}) => {
...
+        documentUploadConfirmation.verifyNoDocumentError(documentUploadConfirmationCopy)
+      })
 
-    it('should return file size too large message for doc', async () => {
-      goToPassportUploadScreen()
-      documentUpload.getUploadInput()
-      documentUpload.upload('over_10mb_face.jpg')
-      documentUploadConfirmation.verifyFileSizeTooLargeError(documentUploadConfirmationCopy)
-    })
+      it('should upload a document after retrying', async () => {
```
Comment (kopijunkie) [inline test/js/specs/happy.js]: Minor: Think this test description should say "should upload a document on retry"

File: test/js/specs/happy.js [hunk 2]
```diff
@@ -70,151 +71,153 @@ describe('Happy Paths', options, ({driver, pageObjects}) => {
...
+        documentUploadConfirmation.verifyFileSizeTooLargeError(documentUploadConfirmationCopy)
+      })
 
-    it('should return unsupported file type error for selfie', async () => {
-      goToPassportUploadScreen(`?async=false&language=&useWebcam=false`)
-      uploadFileAndClickConfirmButton('passport.jpg')
-      uploadFileAndClickConfirmButton('national_identity_card.pdf')
-      documentUploadConfirmation.verifyUnsuppoertedFileError(documentUploadConfirmationCopy)
-    })
+      it('should return use another file type message', async () => {
```
[the remark this answers is not in the document]
Comment (bartzet) [reply test/js/specs/happy.js]: The message is `File not uploading. Try using another file type.`

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://732-pr-onfido-sdk-ui-onfido.surge.sh

File: test/js/pageobjects/Common.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import Base from './BasePage.js'
+import {locale} from '../utils/mochaw'
+
+class Common extends Base {
```
Comment (rfreitas) [inline test/js/pageobjects/Common.js]: how is `Common` different from `Base`?

Comment (bartzet) [reply test/js/pageobjects/Common.js]: The idea for the `Common` class is to have the page objects and functions that are shared across various screens e.g. `verifyTitle` or `backArrow` so we avoid repetitions.

Comment (rfreitas) [reply test/js/pageobjects/Common.js]: but is it different from `Base`? are page objects which might be `Base` but not `Common`?

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-732-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://732-pr-onfido-sdk-ui-onfido.surge.sh

File: test/js/pageobjects/Common.js (hunk 3 above)
Comment (bartzet) [reply test/js/pageobjects/Common.js]: Shall I address it in separate PR? 

@stefaniacardenas already ran the acceptance on that PR so it's ready to be merged. There is also PR for cleaning the Ruby dependencies on top of that branch I won't need to solve conflicts there.

After discussing I understood the changes that would need to be done are:
1. Rename the Common class to Screen
2. Screen extends from Base
3. Other classes in `pageobjects` extend from Screen
4. Move `copy(lang) { return locale(lang) }` to Screen class and remove it from other pageobject classes.

Comment (bartzet) [reply test/js/pageobjects/Common.js]: I have the stash changes ready to for the above but I don't want to introduce that many changes into the PR that was already reviewed and accepted.

**Pull request merged**